ppp (many updates in one)

joeflack4 opened the pull request.

joeflack4 commented on pmix/ppp/odkprompt.py:
```diff
@@ -1,5 +1,7 @@
 """Module for the OdkPrompt class."""
 import textwrap
+from pmix.ppp.constants import MEDIA_FIELDS, LANGUAGE_DEPENDENT_FIELDS, \
```
Moved re-usable values from OdkPrompt class to constants in constants module.

joeflack4 commented on pmix/ppp/odkprompt.py:
```diff
@@ -128,10 +118,11 @@ def reformat_default_lang_vars(row, lang):
         Returns:
             dict: Reformatted representation.
         """
```
Changed logic to manipulate 'row.copy()', as 'row' was not meant to be mutated. Updated logic with respect to class attributes to shared constants refactoring.

*(Note: Just clicking 'start a review' here because I've never done that before and I just want to see what it's like to do that.)*

joeflack4 commented on pmix/ppp/odkform.py:
```diff
@@ -1,11 +1,12 @@
 """Module for the OdkForm class."""
 import datetime
 import os.path
```
Changed logic to manipulate 'row.copy()', as 'row' was not meant to be mutated. Updated logic with respect to class attributes to shared constants refactoring.

joeflack4 replied:
Updated logic with respect to class attributes to shared constants refactoring.

joeflack4 commented on pmix/ppp/odkprompt.py:
```diff
@@ -1,5 +1,7 @@
 """Module for the OdkPrompt class."""
 import textwrap
```
Moved re-usable values from OdkPrompt class to constants in constants module.

joeflack4 commented on pmix/ppp/odkprompt.py:
```diff
@@ -164,7 +156,7 @@ def create_additional_media_fields(row, prefix):
 
         for field in fields_to_add:
             row[field] = ''
```
De-linting.

joeflack4 commented on pmix/ppp/odkrepeat.py:
```diff
@@ -12,6 +12,7 @@ class OdkRepeat:
     Attributes:
         opener (dict): A dictionary row representing first row of repeat group.
         data (list): A list of repeat group components.
```
De-linting which was later removed when adding .pydocstyle.

joeflack4 commented on pmix/ppp/odktable.py:
```diff
@@ -12,6 +12,7 @@ class OdkTable:
         header (OdkPrompt): OdkPrompt representing table header.
         contents (list): List of OdkPrompts consisting of table rows.
         in_repeat (bool): Is this table part of a repeat group?
```
De-linting which was later removed when adding .pydocstyle.

joeflack4 commented on pmix/ppp/templates/styles.html:
```diff
@@ -10,6 +10,10 @@
                 border-spacing: 0;
                 table-layout: fixed;
             }
```
Experimentation styles commented out to be removed after page break fix.

joeflack4 commented on pmix/ppp/templates/styles.html:
```diff
@@ -26,6 +30,13 @@
                 border-bottom: snow;
             }
             /* Classes - Positioning */
```
Max page width set for aesthetics when viewing as html in browser. Added handlers for some unused ID's that Rijad put in, but no values on them. Can just as well be removed.

joeflack4 commented on pmix/xlsform.py:
```diff
@@ -61,6 +61,11 @@ def settings_language(self):
         return default_language
 
     @property
```
Added property 'survey_languages' for instances when 'form_language' may be too explicit, and language determination from specific worksheet may be more useful to know. Refactored accordingly.

joeflack4 commented on pmix/xlstab.py:
```diff
@@ -46,6 +46,11 @@ def __init__(self, *, data=None, name=None):
         super().__init__(data=data, name=name)
         self.assert_unique_cols()
 
```
Added __repr__ for child class, as was incorrectly reflecting parent class. Caused some confusion during debugging.

joeflack4 commented on requirements.txt:
```diff
@@ -2,9 +2,6 @@ astroid==1.4.9
 isort==4.2.5
 lazy-object-proxy==1.2.2
 mccabe==0.6.1
```
Moved developer-specific dependencies to be grouped together at bottom of file for ease of distinction.

joeflack4 commented on test/test_ppp.py:
```diff
@@ -2,17 +2,37 @@
 # -*- coding: utf-8 -*-
 """Unit tests for PPP package."""
 import unittest
```
Added doctests runner logic to recursively find and run all doctests in designated packages 'TEST_PACKAGES'. Runs either together with unit tests, or by itself via CLI when calling 'test_ppp.py'.

joeflack4 commented on test/test_ppp.py:
```diff
@@ -45,7 +65,7 @@ class OdkPromptTest(unittest.TestCase, PppTest):
     def setUp(self):
         self.data = (
             ({'inputs': {
```
Several refactorings, mostly minor, throughout.

joeflack4 commented on test/test_ppp.py:
```diff
@@ -61,6 +81,25 @@ def test_to_dict(self):
         """Test to_dict method."""
         def test_media_fields_in_prompts():
             """Tests for media fields."""
```
Refactoring for 'too many branches'.

joeflack4 commented on test/test_verbiage.py:
```diff
@@ -26,6 +26,7 @@ class TranslationDictNumberSplitTest(unittest.TestCase):
     """Unit tests for the TranslationDict number splitting."""
 
     def test_yes_split_text(self):
```
De-linting throughout file.

joeflack4 commented on test/test_xlsform_language.py:
```diff
@@ -1,8 +1,9 @@
+"""Tests for Xlsform module."""
```
Minor de-linting of file.

jkpr commented on pmix/ppp/__init__.py:
```diff
@@ -2,48 +2,45 @@
         output_format (str): File format to be output.
-        outfile (str): Path to save converted file.
+        out_file (str): Path to save converted file.
+        **language (str): Language to render form.
```
not a keyword argument

jkpr commented on makefile:
```diff
@@ -1,12 +1,45 @@
 
-.PHONY: lint tags ltags test
+PYTHON_JKP=./env/bin/python3
+PYTHON_JEF=python3
```
Let's put a symbolic link on your machine at `env/bin/python3`. I do not like that you have it as the environment python. It has created issues for us in the past.

jkpr commented on pmix/ppp/odkchoices.py:
```diff
@@ -77,6 +78,16 @@ def name_labels(self, lang):
         Returns:
             list: Choice variable names and associated labels for choice list.
         """
+        # 1. 'label' / all language fields alone needs to be supported if no
```
Let's rework `labels`, `name_labels`, and `choice_langs`. There is a lot of room for simplification.

jkpr commented on pmix/ppp/odkform.py:
```diff
@@ -1,31 +1,35 @@
 """Module for the OdkForm class."""
 import datetime
 import os.path
+# from pmix.ppp.constants import LANGUAGE_PERTINENT_WORKSHEETS, \
```
In general: remove commented code.

jkpr commented on pmix/ppp/odkform.py:
```diff
@@ -1,31 +1,35 @@
+
     """
 
     def __init__(self, file=None, wb=None):
```
confusing signature. let's break this into two functions: `__init__(wb)` and `from_file(path)` (the second being a classmethod)

jkpr commented on pmix/ppp/odkform.py:
```diff
@@ -89,10 +98,10 @@ def get_settings(wb):
 
         Args:
-            wb (Workbook): A workbook object representing ODK form.
+            wb (Xlsform): A workbook object representing ODK form.
```
Refactor to remove. Already have `Xlsform().settings`

joeflack4 replied:
Done~

jkpr commented on pmix/ppp/odkform.py:
```diff
@@ -146,45 +155,64 @@ def get_choices(wb, ws):
     @staticmethod
-    def get_languages(wb):
-        """Get survey languages.
+    # def get_language(self, requested_lang, settings, wb):
```
remove

jkpr commented on pmix/ppp/odkform.py:
```diff
@@ -198,30 +226,35 @@ def get_running_conversion_time(self):
+        # lang = kwargs['lang'] if 'lang' in kwargs \
+        #     else self.languages['default_language']
+        # requested_lang = kwargs['lang'] if 'lang' in kwargs else None
+        language = self.get_language(requested_lang=language,
```
we now have a better way of doing this. `if not language: language=self.survey_language`

jkpr commented on pmix/ppp/odkform.py:
```diff
@@ -146,45 +155,64 @@ def get_choices(wb, ws):
+
+        return determined_language
+
+    # Currently unused.
```
Anything related to timing, move into a wrapper (if even needed).

joeflack4 commented:
### Requested Changes Follow-up Summary

- Minor Commits
  - Removed OdkForm.get_language()
  - OdkForm.settings - Taken from Xlsform.settings
  - Conversion Time - Feature removed
  - Makefile
    - Use only one common Python environment path.
    - DRYness, aliases, file organization.
- De-linted.
- Removed `output/` from commit history.
- Comments - Updates and deletions.
